pkg/apis/kubeone/v1beta1/types.go

```diff
@@ -0,0 +1,331 @@
+
+// DigitalOceanSpec defines the DigitalOcean cloud provider
+type DigitalOceanSpec struct{}
+
+// GCESpec defines the GCE cloud provider
+type GCESpec struct{}
+
+// HetznerSpec defines the Hetzner cloud provider
+type HetznerSpec struct{}
+
+// OpenstackSpec defines the Openstack provider
+type OpenstackSpec struct{}
+
+// PacketSpec defines the Packet cloud provider
+type PacketSpec struct{}
+
+// VsphereSpec defines the vSphere provider
+type VsphereSpec struct{}
+
+// NoneSpec defines a none provider
+type NoneSpec struct{}
+
+// VersionConfig describes the versions of components that are installed on the machines
+type VersionConfig struct {
+	Kubernetes string `json:"kubernetes"`
+}
+
+// ClusterNetworkConfig describes the cluster network
+type ClusterNetworkConfig struct {
+	PodSubnet         string `json:"podSubnet"`
+	ServiceSubnet     string `json:"serviceSubnet"`
+	ServiceDomainName string `json:"serviceDomainName"`
+	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+
+	// +optional
+	NetworkID string `json:"networkID,omitempty"`
+}
+
+// CNI config. Only one CNI provider must be used at the single time.
+type CNI struct {
+	Canal    *CanalSpec       `json:"canal"`
+	WeaveNet *WeaveNetSpec    `json:"weaveNet"`
+	External *ExternalCNISpec `json:"external"`
+}
+
+// CanalSpec defines the Canal CNI plugin
+type CanalSpec struct{}
+
+// WeaveNetSpec defines the WeaveNet CNI plugin
+type WeaveNetSpec struct {
+	Encrypted bool `json:"encrypted"`
+}
+
+// ExternalCNISpec defines the external CNI plugin.
+// It's up to the user's responsibility to deploy the external CNI plugin manually or as an addon
+type ExternalCNISpec struct{}
+
+// ProxyConfig configures proxy for the Docker daemon and is used by KubeOne scripts
+type ProxyConfig struct {
+	HTTP    string `json:"http"`
+	HTTPS   string `json:"https"`
+	NoProxy string `json:"noProxy"`
+}
+
+// DynamicWorkerConfig describes a set of worker machines
+type DynamicWorkerConfig struct {
+	Name     string       `json:"name"`
+	Replicas *int         `json:"replicas"`
+	Config   ProviderSpec `json:"providerSpec"`
+}
+
+// ProviderSpec describes a worker node
+type ProviderSpec struct {
+	CloudProviderSpec   json.RawMessage   `json:"cloudProviderSpec"`
+	Labels              map[string]string `json:"labels"`
+	Taints              []corev1.Taint    `json:"taints,omitempty"`
+	SSHPublicKeys       []string          `json:"sshPublicKeys"`
+	OperatingSystem     string            `json:"operatingSystem"`
+	OperatingSystemSpec json.RawMessage   `json:"operatingSystemSpec"`
+
+	// +optional
+	Network *NetworkConfig `json:"network,omitempty"`
+
+	// +optional
+	OverwriteCloudConfig *string `json:"overwriteCloudConfig,omitempty"`
+}
+
+// DNSConfig contains a machine's DNS configuration
+type DNSConfig struct {
+	Servers []string `json:"servers"`
+}
+
+// NetworkConfig contains a machine's static network configuration
+type NetworkConfig struct {
```

Comment:
```suggestion
	Network *ProviderStaticNetworkConfig `json:"network,omitempty"`

	// +optional
	OverwriteCloudConfig *string `json:"overwriteCloudConfig,omitempty"`
}

// DNSConfig contains a machine's DNS configuration
type DNSConfig struct {
	Servers []string `json:"servers"`
}

// ProviderStaticNetworkConfig contains a machine's static network configuration
type ProviderStaticNetworkConfig struct {
```

Reply:
This diff got a bit messy, but as far as I understand, the proposal is to change `NetworkConfig` to `ProviderStaticNetworkConfig`. Is there anything else that I'm missing?

Reply:
nothing more :)